src/audio/dai-legacy.c

```diff
@@ -128,32 +127,13 @@ static void dai_dma_cb(void *arg, enum notify_id type, void *data)
 
 	local_buf = buffer_acquire(dd->local_buffer);
 
-	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
-		ret = dma_buffer_copy_to(local_buf, dma_buf,
-					 dd->process, bytes);
-	} else {
-		ret = dma_buffer_copy_from(dma_buf, local_buf,
-					   dd->process, bytes);
-	}
+	if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
+		dma_buffer_copy_to(local_buf, dma_buf, dd->process, bytes);
+	else
+		dma_buffer_copy_from(dma_buf, local_buf, dd->process, bytes);
 
-	/* assert dma_buffer_copy succeed */
-	if (ret < 0) {
-		struct comp_buffer __sparse_cache *source_c, *sink_c;
-
-		source_c = dev->direction == SOF_IPC_STREAM_PLAYBACK ?
-					local_buf : dma_buf;
-		sink_c = dev->direction == SOF_IPC_STREAM_PLAYBACK ?
-					dma_buf : local_buf;
-		comp_err(dev, "dai_dma_cb() dma buffer copy failed, dir %d bytes %d avail %d free %d",
-			 dev->direction, bytes,
-			 audio_stream_get_avail_samples(&source_c->stream) *
-				audio_stream_frame_bytes(&source_c->stream),
-			 audio_stream_get_free_samples(&sink_c->stream) *
-				audio_stream_frame_bytes(&sink_c->stream));
-	} else {
-		/* update host position (in bytes offset) for drivers */
-		dd->total_data_processed += bytes;
-	}
+	/* update host position (in bytes offset) for drivers */
+	dd->total_data_processed += bytes;
```

Comment:
I'm not sure we can remove this, the DMA copy might fail and we'd need to handle this.

Reply:
it is just perform pcm converter function instead of dma operation, dma operation happend in other place, search "dma_reload", there is error handling.